core/blockchain_impl.go

```diff
@@ -2367,8 +2370,15 @@ func (bc *BlockChainImpl) DeleteCrossLinks(cls []types.CrossLink) error {
 
 func (bc *BlockChainImpl) ReadCrossLink(shardID uint32, blockNum uint64) (*types.CrossLink, error) {
 	bytes, err := rawdb.ReadCrossLinkShardBlock(bc.db, shardID, blockNum)
-	if err != nil {
+	if err == goleveldb.ErrNotFound {
+		utils.Logger().Debug().Msgf("ReadCrossLink: not found")
+		return nil, ErrCrosslinkNotFound
+	} else if err != nil {
+		utils.Logger().Debug().Err(err).Msgf("ReadCrossLink failed")
 		return nil, err
+	} else if len(bytes) == 0 {
```

Comment:
Q: if that case ever happened, what would be the fix ? delete shard 0 DB and resync ?

Reply:
This indicates that the database is not returning an error but is returning nil bytes for the crosslink. If this occurs, it suggests the crosslink is corrupted or not stored properly in the database. Resyncing the database should resolve the issue.